src/marshmallow/utils.py

```diff
@@ -197,7 +197,12 @@ def from_timestamp(value: typing.Any) -> dt.datetime:
 
     # Load a timestamp with utc as timezone to prevent using system timezone.
     # Then set timezone to None, to let the Field handle adding timezone info.
-    return dt.datetime.fromtimestamp(value, tz=dt.timezone.utc).replace(tzinfo=None)
+    try:
+        return dt.datetime.fromtimestamp(value, tz=dt.timezone.utc).replace(tzinfo=None)
+    except OverflowError as exc:
+        raise ValueError("Timestamp is too large") from exc
+    except OSError as exc:
+        raise ValueError("Failed to convert this value to datetime") from exc
```

Comment:
The following sentence might be slightly better and more in line with other errors:

> Error converting value to datetime

Just my feeling. I'm not a native English speaker.

Reply:
Updated.